Sources/Swift/Tools/Batcher/BatchStorage.swift

```diff
@@ -0,0 +1,10 @@
+protocol BatchStorage<Element> {
+    associatedtype Element
+
+    mutating func append(_ element: Element) throws
```

Comment:
`l`: I think we're usually adding to the buffers. Append kind of suggests we add it to the end of the buffer or something, but that's not always the case. What if we use a ring buffer internally. 
```suggestion
    mutating func add(_ element: Element) throws
```